src/reducers/secrets.js

```diff
@@ -27,12 +27,24 @@ function byNamespace(state = {}, action) {
         });
       }, {});
       return merge({}, state, namespaces);
-    case 'SECRET_DELETE_SUCCESS':
+    case 'SecretCreated':
+    case 'SecretUpdated':
+      const secret = {
+        [action.payload.metadata.namespace]: {
+          [action.payload.metadata.name]: {
+            name: action.payload.metadata.name,
+            namespace: action.payload.metadata.namespace,
+            annotations: action.payload.metadata.annotations
+          }
+        }
+      };
+      return merge({}, state, secret);
+    case 'SecretDeleted':
       const newState = state;
```

Comment:
I didn't notice this before but this code is actually modifying the current state, instead of operating on a copy... Since `newState` is simply a reference to `state`, calling `delete newState[namespace][name]` removes this item from `state`. 

When working with redux it's important to ensure we always treat the current state as immutable, only operating on copies / building new state.